### PR TITLE
add banned user scenario to get and list workspaces tests

### PR DESCRIPTION
**Pull request by mfrancisc**
This PR add test scenarios in which a banned user cannot get and list workspaces.

It is related to this PR: https://github.com/codeready-toolchain/toolchain-e2e/pull/1036 which adds tests for banned users in proxy.

### CODE REVIEW

File: test/e2e/parallel/proxy_test.go [hunk 1]
```diff
@@ -969,6 +984,24 @@ func TestSpaceLister(t *testing.T) {
...
+		assert.Empty(t, workspace)
+	})
+
+	t.Run("banned user cannot list workspaces", func(t *testing.T) {
+		// when
+		workspaces := users["banneduser"].listWorkspaces(t, hostAwait)
+
+		// then
+		// banned user should not see any workspace
+		require.Len(t, workspaces, 0)
```
Comment (filariow) [inline test/e2e/parallel/proxy_test.go:1002]:
```suggestion
		require.Empty(t, workspaces)
```

File: test/e2e/parallel/proxy_test.go [hunk 2]
```diff
@@ -969,6 +984,24 @@ func TestSpaceLister(t *testing.T) {
 		require.NoError(t, err)
 	})
 
+	t.Run("banned user cannot get workspace", func(t *testing.T) {
+		// when
+		workspace, err := users["banneduser"].getWorkspace(t, hostAwait, users["banneduser"].compliantUsername)
+
+		// then
+		require.EqualError(t, err, "the server could not find the requested resource (get workspaces.toolchain.dev.openshift.com banneduser)")
+		assert.Empty(t, workspace)
```
Comment (filariow) [inline test/e2e/parallel/proxy_test.go:993]: shouldn't we use `require.Empty` instead?

```suggestion
		require.Empty(t, workspace)
```

**Review by filariow — COMMENTED** [review]
except for Linter's complaint and another minor suggestion, it lgtm

File: test/e2e/parallel/proxy_test.go (hunk 2 above)
Comment (filariow) [reply test/e2e/parallel/proxy_test.go:993]: Or `require.Nil`?
```suggestion
		require.Nil(t, workspace)
```

**Review by alexeykazakov — APPROVED** [review]
Nice.

File: test/e2e/parallel/proxy_test.go (hunk 1 above)
Comment (mfrancisc) [reply test/e2e/parallel/proxy_test.go:1002]: ty.

File: test/e2e/parallel/proxy_test.go (hunk 2 above)
Comment (mfrancisc) [reply test/e2e/parallel/proxy_test.go:993]: thanks!

File: test/e2e/parallel/proxy_test.go [hunk 3]
```diff
@@ -969,6 +984,24 @@ func TestSpaceLister(t *testing.T) {
 		require.NoError(t, err)
 	})
 
+	t.Run("banned user cannot get workspace", func(t *testing.T) {
+		// when
+		workspace, err := users["banneduser"].getWorkspace(t, hostAwait, users["banneduser"].compliantUsername)
+
+		// then
+		require.EqualError(t, err, "the server could not find the requested resource (get workspaces.toolchain.dev.openshift.com banneduser)")
```
Comment (filariow) [inline test/e2e/parallel/proxy_test.go:992]: Is it correct that it's returning an error on banneduser? I'd expect either a Forbidden or a NoMatch on workspaces

File: test/e2e/parallel/proxy_test.go [hunk 4]
```diff
@@ -969,6 +984,24 @@ func TestSpaceLister(t *testing.T) {
 		require.NoError(t, err)
 	})
 
+	t.Run("banned user cannot get workspace", func(t *testing.T) {
+		// when
+		workspace, err := users["banneduser"].getWorkspace(t, hostAwait, users["banneduser"].compliantUsername)
```
[the remark this answers is not in the document]
Comment (mfrancisc) [reply test/e2e/parallel/proxy_test.go]: thanks for pointing that out @rsoaresd . I think it's enough to check that the proxy client initialization fails when the user is banned. I've updated the test.

File: test/e2e/parallel/proxy_test.go (hunk 3 above)
Comment (mfrancisc) [reply test/e2e/parallel/proxy_test.go]: actually it looks like there should be an error returned when the proxy is initialized, as pointed out https://github.com/codeready-toolchain/toolchain-e2e/pull/1037#discussion_r1723049502 . 

I've updated the test to only try and initialize the proxy client for the banned user, which will result in an error.

**Review by MatousJobanek — APPROVED** [review]

**Pull request merged**